Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/DownloadConfigMapper.kt

```diff
@@ -2,20 +2,93 @@ package de.rki.coronawarnapp.appconfig.mapping
 
 import dagger.Reusable
 import de.rki.coronawarnapp.appconfig.KeyDownloadConfig
+import de.rki.coronawarnapp.diagnosiskeys.server.LocationCode
 import de.rki.coronawarnapp.server.protocols.internal.AppConfig
-import de.rki.coronawarnapp.server.protocols.internal.KeyDownloadParameters
+import de.rki.coronawarnapp.server.protocols.internal.KeyDownloadParameters.KeyDownloadParametersAndroid
+import org.joda.time.Duration
+import org.joda.time.LocalDate
+import org.joda.time.LocalTime
+import org.joda.time.format.DateTimeFormat
+import timber.log.Timber
 import javax.inject.Inject
 
 @Reusable
 class DownloadConfigMapper @Inject constructor() : KeyDownloadConfig.Mapper {
     override fun map(rawConfig: AppConfig.ApplicationConfiguration): KeyDownloadConfig {
+        val rawParameters = rawConfig.androidKeyDownloadParameters
 
         return KeyDownloadConfigContainer(
-            keyDownloadParameters = rawConfig.androidKeyDownloadParameters
+            individualDownloadTimeout = rawParameters.individualTimeout(),
+            overallDownloadTimeout = rawParameters.overAllTimeout(),
+            invalidDayETags = rawParameters.mapDayEtags(),
+            invalidHourEtags = rawParameters.mapHourEtags()
         )
     }
 
+    private fun KeyDownloadParametersAndroid.individualTimeout(): Duration {
+        return if (downloadTimeoutInSeconds == 0) Duration.standardSeconds(60)
```

Comment:
Do we need these default/fallback values here?

Reply:
A protobuf config value that is missing defaults to `0` for numbers, right? Unless we want different behavior for "missing value", I'd like to always map it to sane defaults.

Reply:
In other cases I'm especially mapping it to fallbacks to prevent "division by 0 exceptions".

Reply:
I was wondering the same thing when reviewing and I agree with the overwrite. Maybe the function name could hint that 0 is not a valid value (will be replaced)